src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -467,14 +467,30 @@ void Replayer<I>::handle_load_remote_group_snapshots(int r) {
   if (is_replay_interrupted()) {
     return;
   }
-
-  auto last_remote_snap = m_remote_group_snaps.rbegin();
   if (r < 0) {  // may be remote group is deleted?
     derr << "error listing remote mirror group snapshots: " << cpp_strerror(r)
          << dendl;
     handle_replay_complete(r, "failed to list remote group snapshots");
     return;
-  } else if (is_resync_requested()) {
+  }
+  for (auto snap = m_remote_group_snaps.begin(); snap!= m_remote_group_snaps.end();) {
+    auto ns = std::get_if<cls::rbd::GroupSnapshotNamespaceMirror>(&snap->snapshot_namespace);
+    if (ns == nullptr) {
+      ++snap;
+      continue;
+    }
+    if (ns->mirror_peer_uuids.count(m_remote_mirror_peer_uuid) == 0) {
+      dout(10) << "Removing snapshot " << snap->id
```

Comment:
1. start the long lines with small case?
2. Instead of using `Removing` may be use `filtering out`, for me removing gives me a feel of `pruning` here.